provisionersdk/archive.go

```diff
@@ -66,14 +89,20 @@ func Tar(w io.Writer, logger slog.Logger, directory string, limit int64) error {
 		)
 	}
 
-	err = filepath.Walk(directory, func(file string, fileInfo os.FileInfo, err error) error {
+	err = afero.Walk(afs, directory, func(file string, fileInfo os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
 		var link string
 		if fileInfo.Mode()&os.ModeSymlink == os.ModeSymlink {
-			link, err = os.Readlink(file)
-			if err != nil {
+			link, err = afs.ReadlinkIfPossible(file)
+			if errors.Is(err, afero.ErrNoReadlink) {
+				// It's unclear how we could hit this code path -- if the underlying FS implementation
+				// does not support symbolic links, how could there exist a symlink on the FS?
+				// Instead of blocking though, just log the error and continue without resolving the
+				// symlink.
+				logger.Error(context.Background(), "developer error: fs %T should support reading symlinks yet still returned %w", afs, err)
```

Comment:
review: I'm also open to returning an error here; I'm taking the cowardly route for now and just logging an error.

Reply:
I think this is the more reasonable approach here 👍